File: macos_app/NagarikPatro/CalendarCore/Sources/CalendarCore/BsDate.swift
import Foundation

/// A date in the Bikram Sambat (BS) calendar system used in Nepal.
public struct BsDate: Equatable, Hashable, Comparable, Sendable {
    public let year: Int
    public let month: Int  // 1-12
    public let day: Int    // 1-32

    public init(year: Int, month: Int, day: Int) {
        self.year = year
        self.month = month
        self.day = day
    }

    public static func < (lhs: BsDate, rhs: BsDate) -> Bool {
        if lhs.year != rhs.year { return lhs.year < rhs.year }
        if lhs.month != rhs.month { return lhs.month < rhs.month }
        return lhs.day < rhs.day
    }
}

File: windows_app/NagarikPatro/Models/BsDate.cs
namespace NagarikPatro.Models;

/// <summary>
/// A date in the Bikram Sambat (BS) calendar system used in Nepal.
/// </summary>
public readonly record struct BsDate(int Year, int Month, int Day) : IComparable<BsDate>
{
    public int CompareTo(BsDate other)
    {
        int cmp = Year.CompareTo(other.Year);
        if (cmp != 0) return cmp;
        cmp = Month.CompareTo(other.Month);
        if (cmp != 0) return cmp;
        return Day.CompareTo(other.Day);
    }

    public override string ToString() => $"{Year}-{Month:D2}-{Day:D2}";
}
